internal/cephfs/nodeserver.go

```diff
@@ -403,22 +410,29 @@ func (ns *NodeServer) NodeGetCapabilities(
 func (ns *NodeServer) NodeGetVolumeStats(
 	ctx context.Context,
 	req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	var err error
-	targetPath := req.GetVolumePath()
-	if targetPath == "" {
-		err = fmt.Errorf("targetpath %v is empty", targetPath)
-
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+	// Get volume path
+	// With Kubernetes version>=v1.19.0, stats request carries volume_path and
+	// staging_target_path, what csi requires is staging_target_path.
+	volumePath := req.GetStagingTargetPath()
```

Comment:
In fact I see it exist since v1.17.0 as per git history

```
[0] pkalever 😎 kubernetes✨ git contain 72bcec4e4d516dcac227606541f990af13fb2918 | grep v1.17.0
v1.17.0
v1.17.0-beta.2
v1.17.0-rc.1
v1.17.0-rc.2
```

Reply:
>I remember we used similar stuff here https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/nodeserver.go#L990

You are referring to NodeExpandVolume, not NodeGetVolumeStats?

Reply:
Sure, lets use VolumePath because kubelet still uses volumePath in its NodeGetVolumeStatsRequest.